modules/gapi/src/backends/fluid/gfluidimgproc.cpp

```diff
@@ -2132,11 +2135,29 @@ CV_ALWAYS_INLINE void calcRowLinear(const cv::gapi::fluid::View& in,
     {
         auto index0 = mapsy[outY + l] - inY;
         auto index1 = mapsy[outSz.height + outY + l] - inY;
+
         src0[l] = in.InLine<const float>(index0);
         src1[l] = in.InLine<const float>(index1);
         dst[l] = out.OutLine<float>(l);
     }
 
+#if CV_AVX2
+    constexpr int nlanes = 8;
+
+    if (inSz.width >= nlanes && outSz.width >= nlanes)
+    {
+        avx2::calcRowLinear32FC1Impl(reinterpret_cast<float**>(dst),
+                                     reinterpret_cast<const float**>(src0),
```

Comment:
reinterpret cast is normally a red flag, can this be avoided? can `avx2::calcRowLinear32FC1Impl` also accept a `float*[]` type instead?

Reply:
Ok. Removed.